src/instruction/instructions/sbcs.ts

```diff
@@ -0,0 +1,50 @@
+import { Halfword, Word } from 'types/binary'
+
+import { Flag, Registers } from 'board/registers'
+import { IMemory } from 'board/memory/interfaces'
+import { add } from 'board/alu'
+
+import { ILabelOffsets } from 'instruction/interfaces'
+import {
+  checkOptionCount,
+  create,
+  createLowRegisterBits,
+  getBits,
+  setBits
+} from 'instruction/opcode'
+
+import { BaseInstruction } from './base'
+
+/**
+ * Represents a 'SBCS' instruction which does a substraction with carry.
+ */
+export class SbcsInstruction extends BaseInstruction {
+  public name: string = 'SBCS'
+  public pattern: string = '0100000110XXXXXX'
+  private rdnPattern: string = '0100000110000XXX'
+  private rmPattern: string = '0100000110XXX000'
+
+  public encodeInstruction(options: string[], labels: ILabelOffsets): Halfword {
+    checkOptionCount(options, 2)
+    let opcode: Halfword = create(this.pattern)
+    opcode = setBits(opcode, this.rdnPattern, createLowRegisterBits(options[0]))
+    opcode = setBits(opcode, this.rmPattern, createLowRegisterBits(options[1]))
+    return opcode
+  }
+
+  public executeInstruction(
+    opcode: Halfword,
+    registers: Registers,
+    memory: IMemory
+  ): void {
+    const rdn = getBits(opcode, this.rdnPattern)
+    const rm = getBits(opcode, this.rmPattern)
+    const carry = registers.isFlagSet(Flag.C) ? 1 : 0
+    const notRm = Word.fromUnsignedInteger(
+      ~registers.readRegister(rm.value).value >>> 0
+    )
+    const result = add(registers.readRegister(rdn.value), notRm.add(carry))
```

Comment:
same as for adcs.ts when carry is set and notRm has certain values an overflow or a carry can already happen on carry add.

Why is not the sub method used of alu?

According to cheat sheet carry should be inverted and not rm:
![image](https://user-images.githubusercontent.com/56474048/162071765-e54871a2-e5f8-430f-ad75-4e5d640ae8af.png)

Reply:
According to the CT1 slides, both implementations are possible:
![image](https://user-images.githubusercontent.com/15670381/162379151-533436c0-d7f1-4ae0-aa70-e6f49153ea58.png)

Reply:
okey, nevertheless if Rm register value is all 0 then not of it is 0xffffffff and if carry is set then on add there will be an overflow on add of carry or when Rm valu is 0x80000000 then inverted is 0x7fffffff and on adding carry an overflow happens

Reply:
I've now improved the implementation and added some tests as well.